src/pp_ast.mli

```diff
@@ -60,28 +60,46 @@ module Config : sig
           be. *)
 end
 
+type simple_val =
+  | Unit
+  | Int of int
+  | String of string
+  | Bool of bool
+  | Char of char
+  | Array of simple_val list
+  | Float of float
+  | Int32 of int32
+  | Int64 of int64
+  | Nativeint of nativeint
+  | Record of (string * simple_val) list
+  | Constr of string * simple_val list
+  | Tuple of simple_val list
+  | List of simple_val list
+  | Special of string
+
+type printer = Format.formatter -> simple_val -> unit
 type 'a pp = Format.formatter -> 'a -> unit
-type 'a configurable = ?config:Config.t -> 'a pp
+type 'a configurable = ?config:Config.t -> ?printer:printer -> 'a pp
```

Comment:
For future reference: this could be considered a breaking change here.

Do you think it would make sense to make the `?printer` a part of the config directly? That way this functions API does not change and we just add a new optional argument to `Config.make`.

Reply:
We could eventually split the full config into a lifter config and a printing config.

Reply:
> For future reference: this could be considered a breaking change here.

I thought that adding this feature with backward compatibility would avoid that to be considered breaking change 😞 

Adding it to the Config like this couldn't also be considered a breaking change, but for the Config? 
<img width="585" alt="image" src="https://github.com/user-attachments/assets/0b020a69-e789-42cf-8e47-cc9c74ea89a6" />

Reply:
I think `Config.make` will pretty much always be applied fully as opposed to the printer which could be partially applied with the `?config` argument so it can then be used as a regular `'a pp`.

You are right that strictly speaking it is the same kind of change but given the nature of the two functions, I think it's far less likely to break user code that way.

What do you think?

Reply:
That makes sense to me